### PR TITLE
feat: implement signals

### PR DESCRIPTION
**Pull request by kurtaking**
### Summary 📚

Announcements plugins now supports signals.

### Details 🔬 

When an announcement is created, a real-time update will occur in the banner if your backstage instance supports signals.

### Checklist ✅ 

* [x] I have updated the necessary documentation
* [x] I have signed off all my commits as required by [DCO](https://GitHub.com/apps/dco/)
* [x] My build is green

### CODE REVIEW

File: plugins/announcements/package.json [hunk 1]
```diff
@@ -63,6 +63,8 @@
     "@backstage/plugin-permission-react": "^0.4.26",
     "@backstage/plugin-search-common": "^1.2.14",
     "@backstage/plugin-search-react": "^1.8.0",
+    "@backstage/plugin-signals": "^0.0.10",
```
Comment (drodil) [inline plugins/announcements/package.json:66]: Frotend plugin should not depend to another frontend plugin. This should be moved to `devDependencies`

File: plugins/announcements/src/components/NewAnnouncementBanner/NewAnnouncementBanner.tsx [hunk 2]
```diff
@@ -152,3 +157,33 @@ export const NewAnnouncementBanner = (props: NewAnnouncementBannerProps) => {
     </>
   );
 };
+
+export const SignalAnnouncementBanner = (props: NewAnnouncementBannerProps) => {
```
Comment (drodil) [inline plugins/announcements/src/components/NewAnnouncementBanner/NewAnnouncementBanner.tsx:161]: I am not sure how this is supposed to be used? I think signals could have been integrated to the existing functionality?

Comment (kurtaking) [reply plugins/announcements/src/components/NewAnnouncementBanner/NewAnnouncementBanner.tsx:161]: I agree, and should have mentioned this was a placeholder until I could at least get the plugins installed and working correctly.

**Review by drodil — APPROVED** [review]
Looking good!

**Comment by kurtaking** [comment]
> Looking good!

Thank you for the help and review! 🥇

**Pull request merged**